benches/flamegraph.rs

```diff
@@ -0,0 +1,37 @@
+extern crate criterion;
+extern crate inferno;
+
+use criterion::*;
+use inferno::flamegraph;
+use std::fs::File;
+use std::io::{BufReader, Cursor, Read};
+
+fn flamegraph_benchmark(c: &mut Criterion) {
+    let mut f = File::open("tests/data/collapse-perf/results/example-perf-stacks-collapsed.txt")
+        .expect("file not found");
+
+    let mut bytes = Vec::new();
+    f.read_to_end(&mut bytes).expect("Could not read file");
+
+    let len = bytes.len();
```

Comment:
This is the size of the input, not the output, yet you're passing it to `with_capacity` for `result`. That doesn't seem right?

Reply:
Yeah, it doesn't seem right. I copied this from the collapse benchmarks. I can fix it in those, too. I'll just use `Vec::new()` for `result`.

Reply:
So, I'm not quite sure that's what we want to do either, because it means we'll also be measuring a bunch of re-allocations of that `Vec` as it grows. I think the trick is going to use `fs::metadata().len()` on a location that we know is the resulting, folded file.

Reply:
Even better, why not just use `io::sink()` for the result since we don't even need the result anyway?

Reply:
That's a great idea! Let's do the same in the others too.